### PR TITLE
 Duration Input Component

### LINKED ISSUE #806: Duration input component
We have a date time picker right now where we work out the number of seconds until that time but we also need a duration input for better UX in specific cases. For example with vesting contracts we specify time until cliff, this makes sense to sue this component. However with the existing date time that would make sense for choosing specific end dates for inputs. 

We want a component styled the same as the time input of current react-datetime but with number inputs horizontally for:
Years | Months | Days | Hours | Minutes | Seconds
This should return the time in seconds

### PR DESCRIPTION
**Pull request by hhamud**
# Description

Closes #806 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

**Comment by rossneilson** [comment]
I should have been clearer in the issue but can we add it to the generic call form for components matching the `time` component value so we can test this in the UI

File: src/components/Guilds/common/Form/DurationInput/WarningInput.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { Flex } from 'Components/Primitives/Layout';
+import styled from 'styled-components';
+
+interface WarningInputProps {
+  timeColumn: string;
+  value: string;
+  limit: any;
```
Comment (MiltonTulli) [inline src/components/Guilds/common/Form/DurationInput/WarningInput.tsx:7]: Can we add types for this? 
```
limit: { max: number; min: number }
```

File: src/components/Guilds/common/Form/DurationInput/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
+import { cleanup, screen, waitFor } from '@testing-library/react';
+import { renderHook, act } from '@testing-library/react-hooks';
+import '@testing-library/jest-dom';
```
Comment (MiltonTulli) [inline src/components/Guilds/common/Form/DurationInput/index.test.tsx:3]: No need to import this. If you have issues, try updating with latest beta v.

File: src/components/Guilds/common/Form/DurationInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import { DURATION_LIMITS } from 'constants/Duration';
+import { useDuration } from 'hooks/Guilds/useDuration';
+import WarningInput from './WarningInput';
+import { Flex } from 'Components/Primitives/Layout';
+import { Button } from 'old-components/common/Button';
+import { Modal } from 'old-components/Guilds/common/Modal';
+import NumericalInput from 'old-components/Guilds/common/Form/NumericalInput';
+
+const Column = styled(Flex)`
+  flex-direction: column;
```
Comment (MiltonTulli) [inline src/components/Guilds/common/Form/DurationInput/index.tsx:13]: Couldn't this be based on a generic column component?

File: src/components/Guilds/common/Form/DurationInput/index.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+})`
+  background-color: transparent;
+  margin: 0;
+  border: none;
+  :hover:enabled {
+    border-color: none;
+  }
+`;
+
+const WarningRow = styled(Flex)``;
```
Comment (MiltonTulli) [inline src/components/Guilds/common/Form/DurationInput/index.tsx:34]: If there is no customization you can delete this and go with just the Flex component.

File: src/components/Guilds/common/Form/DurationInput/index.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  } = useDuration();
+
+  return (
+    <Modal isOpen={isOpen} onDismiss={onDismiss}>
+      <>
+        <Container>
+          {Object.keys(duration).map((value, index) => {
+            const count = duration[value];
+            return (
+              <Column>
```
Comment (MiltonTulli) [inline src/components/Guilds/common/Form/DurationInput/index.tsx:53]: I think you need a mapping key here.

File: src/components/Guilds/common/Form/DurationInput/index.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                  <GoTriangleDown />
+                </ColumnButton>
+              </Column>
+            );
+          })}
+        </Container>
+        <WarningRow>
+          {Object.keys(duration).map(value => {
+            return (
+              <WarningInput
```
Comment (MiltonTulli) [inline src/components/Guilds/common/Form/DurationInput/index.tsx:85]: here too. Please add `key` prop

**Pull request closed without merging**